### PR TITLE
Add network diagrams to README.md

### PR DESCRIPTION
**Pull request by sboeuf**
This PR adds network diagrams describing two different way the network can be setup in virtcontainers, that is either with CNI or with CNM.
It also gives the links to all functions inside the code performing each step of this network setup.

### CODE REVIEW

**Comment by sboeuf** [comment]
@sameo @mrcastel please let me know what you think about this doc.

File: README.md [hunk 1]
```diff
@@ -101,3 +101,77 @@ The function will fail if the Pod is running. In that case `StopPod()` needs to
 
 
 An example tool using the `virtcontainers` API is provided in the `hack/virtc` package.
+
+## Networking
+
+Virtcontainers implements two different way of setting up pod's network:
```
Comment (dlespiau) [inline README.md]: s/way/ways/

File: README.md [hunk 2]
```diff
@@ -101,3 +101,77 @@ The function will fail if the Pod is running. In that case `StopPod()` needs to
...
+
+__Runtime network setup with CNM__
+
+1. Read config.json
+
+2. Create the network namespace ([code](https://github.com/containers/virtcontainers/blob/0.5.0/cnm.go#L108-L120))
+
+3. Call the prestart hook (from inside the netns) ([code](https://github.com/containers/virtcontainers/blob/0.5.0/api.go#L46-L49))
+
+4. Scan network interfaces inside netns and get the name of the interface created by prestart hook ([code](https://github.com/containers/virtcontainers/blob/0.5.0/cnm.go#L70-L106))
```
Comment (dlespiau) [inline README.md]: s/prestart hook/the prestart hook/

**Comment by dlespiau** [comment]
lgtm

**Comment by sameo** [comment]
LGTM

**Pull request merged**